hpo_demo_setup.sh

```diff
@@ -0,0 +1,307 @@
+		## Output format expected for BENCHMARK_OUTPUT is "Objfunc_result=0.007914818407446147 Benchmark_status=success"
+		echo "#######################################"
+		echo
+	        echo "Run the benchmark for trial ${i}"
+		echo
+		BENCHMARK_OUTPUT=$(./hpo_helpers/runbenchmark.sh "hpo_config.json" "${SEARCHSPACE_JSON}" "$i")
+		echo ${BENCHMARK_OUTPUT}
+		obj_result=$(echo ${BENCHMARK_OUTPUT} | cut -d "=" -f2 | cut -d " " -f1)
+		trial_state=$(echo ${BENCHMARK_OUTPUT} | cut -d "=" -f3 | cut -d " " -f1)
+		if [[ ${obj_result} == "" || ${trial_state} == "" ]]; then
+			obj_result=0
+			trial_state="prune"
+		fi
+
+		## Step 4: Send the results of benchmark to HPOaaS
+		echo "#######################################"
+		echo
+        	echo "Send the benchmark results for trial ${i}"
+		curl  -LfSs -H 'Content-Type: application/json' ${URL}/experiment_trials -d '{"experiment_id" : "'"${eid}"'", "trial_number": '"${i}"', "trial_result": "'"${trial_state}"'", "result_value_type": "double", "result_value": '"${obj_result}"', "operation" : "EXP_TRIAL_RESULT"}'
+		check_err "ERROR: Sending the results to HPO failed."
+		echo
+		sleep 5
+		## Step 5 : Generate a subsequent trial
+		if (( i < ${ttrial} - 1 )); then
+			echo "#######################################"
+			echo
+	        	echo "Generate subsequent trial of ${i}"
+			curl  -LfSs -H 'Content-Type: application/json' ${URL}/experiment_trials -d '{"experiment_id" : "'"${eid}"'", "operation" : "EXP_TRIAL_GENERATE_SUBSEQUENT"}'
+			check_err "ERROR: Generating the subsequent trial failed."
+			echo
+		fi
+	done
+
+	echo "#######################################"
+	echo
+	echo "Experiment complete"
+	echo
+
+}
+
+function hpo_start() {
+	
+	# Start all the installs
+	start_time=$(get_date)
+	echo
+	echo "#######################################"
+	echo "#        HPO Demo Setup               #"
+	echo "#######################################"
+	echo
+
+	if [ ${hpo_restart} -eq 0 ]; then
+		clone_repos
+	fi
+	hpo_install
+	sleep 10
+	if [ ${EXPERIMENT_START} -eq 1 ]; then
+		hpo_experiments
+	fi
+	echo
+	end_time=$(get_date)
+	elapsed_time=$(time_diff "${start_time}" "${end_time}")
+	echo "Success! HPO demo setup took ${elapsed_time} seconds"
+	echo
+	echo "Look into experiment-output.csv for configuration and results of all trials"
+	echo
+
+}
+
+function hpo_terminate() {
+	start_time=$(get_date)
+	echo
+	echo "#######################################"
+	echo "#       HPO Demo Terminate       #"
+	echo "#######################################"
+	echo
+	pushd hpo >/dev/null
+                ./deploy_hpo.sh -t -c ${CLUSTER_TYPE}
+		check_err "ERROR: Failed to terminate hpo"
+        popd >/dev/null
+	delete_repos
+	end_time=$(get_date)
+	elapsed_time=$(time_diff "${start_time}" "${end_time}")
+	echo "Success! HPO demo cleanup took ${elapsed_time} seconds"
+	echo
+}
+
+# By default we start the demo and experiment
+hpo_restart=0
+start_demo=1
+HPO_DOCKER_IMAGE=""
```

Comment:
We can set the default docker image for HPO by reading the version number in the file version.py in HPO repo

Reply:
I'm getting the version number in line 113 before starting the server if HPO_DOCKER_IMAGE is not set.